### PR TITLE
EREGCSC-1909 getting by page results in our search

### PR DESCRIPTION
**Pull request by thwalker6**
Resolves # EREGCSC-1909

**Description-**

Urls that had a page number were not appearing properly in the search results

- Search results from search.gov with a page number will now appear in the results.

### CODE REVIEW

File: solution/backend/resources/views/resources.py [hunk 1]
```diff
@@ -109,10 +116,18 @@ def get_queryset(self, urls):
         locations_prefetch = AbstractLocation.objects.all().select_subclasses()
         category_prefetch = AbstractCategory.objects.all().select_subclasses().select_related("subcategory__parent")
 
+        url_filters = [Q(url_annotated__istartswith=url) for url in urls]
+        # gets thes first of the list to sbeing the query set
```
Comment (peggles2) [inline solution/backend/resources/views/resources.py:120]: small typo sbeing change to being.

File: solution/backend/resources/views/resources.py [hunk 2]
```diff
@@ -109,10 +116,18 @@ def get_queryset(self, urls):
...
+        url_query = url_filters.pop()
+
+        # now we are comibning the remaing ones by putting an or statement in between.
```
Comment (peggles2) [inline solution/backend/resources/views/resources.py:123]: just some typos to fix comibning

**Review by peggles2 — CHANGES_REQUESTED** [review]
Looks good, just need to fix the typos.

File: solution/backend/resources/tests/test_viewset_functions.py [hunk 3]
```diff
@@ -69,7 +70,8 @@ def test_append_snippet(self):
         with open("resources/tests/fixtures/url_snippet_dict.json") as f:
             urls = json.load(f)
         ordered_resources = test_view_set.sort_by_url_list(urls.keys(), resources)
-        results = test_view_set.append_snippet(ordered_resources, urls)
+        resources = list(chain.from_iterable(ordered_resources))
+        results = test_view_set.append_snippet(resources, urls)
         num = 1
         for r in results:
             self.assertEqual(f'...site{num} snippet', r.snippet)
```
[the remark this answers is not in the document]
Comment (thwalker6) [reply solution/backend/resources/tests/test_viewset_functions.py:77]: Added a test that does both.

File: solution/backend/resources/views/resources.py (hunk 2 above)
Comment (thwalker6) [reply solution/backend/resources/views/resources.py]: fixed

File: solution/backend/resources/views/resources.py (hunk 1 above)
Comment (thwalker6) [reply solution/backend/resources/views/resources.py]: fixed

**Review by peggles2 — APPROVED** [review]
LGTM 🌮

**Pull request merged**